src/usdb_syncer/db/__init__.py

```diff
@@ -270,33 +271,65 @@ def sql(self) -> str | None:  # noqa: C901
             case SongOrder.PINNED:
                 return "sync_meta.pinned"
             case SongOrder.TXT:
-                return "txt.sync_meta_id IS NULL"
+                return (
+                    "CASE "
+                    "WHEN txt.sync_meta_id IS NULL THEN 0 "
+                    "WHEN txt.status = 'skipped_disabled' THEN 1 "
+                    "WHEN txt.status = 'skipped_unavailable' THEN 2 "
+                    "WHEN txt.status = 'failed' THEN 3 "
+                    "WHEN txt.status = 'failed_existing' THEN 4 "
+                    "WHEN txt.status = 'fallback' THEN 5 "
+                    "WHEN txt.status = 'success' THEN 6 "
+                    "ELSE 7 END"
```

Comment:
Is repeated 4 times instead of using a single f string.
Alternatively, the status could also be stored as a number.